src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -876,15 +876,16 @@ public Task<bool> HasSuccessfullyLoadedAsync(SolutionState solution, Cancellatio
             private AsyncLazy<VersionStamp> _lazyDependentVersion;
             private AsyncLazy<VersionStamp> _lazyDependentSemanticVersion;
 
-            public async Task<VersionStamp> GetDependentVersionAsync(SolutionState solution, CancellationToken cancellationToken)
+            public Task<VersionStamp> GetDependentVersionAsync(SolutionState solution, CancellationToken cancellationToken)
             {
                 if (_lazyDependentVersion == null)
                 {
                     // note: solution is captured here, but it will go away once GetValueAsync executes.
-                    Interlocked.CompareExchange(ref _lazyDependentVersion, new AsyncLazy<VersionStamp>(c => ComputeDependentVersionAsync(solution, c), cacheResult: true), null);
+                    var tmp = solution;
```

Comment:
Why was this necessary here?

Reply:
This avoids a closure allocation for the fast path.

Reply:
Ah, I see. Can you add a comment to that effect?